DigitalLearningSolutions.Web/ViewModels/LearningMenu/ContentViewerViewModel.cs

```diff
@@ -1,50 +1,77 @@
 ﻿namespace DigitalLearningSolutions.Web.ViewModels.LearningMenu
 {
+    using System.Text.RegularExpressions;
+    using DigitalLearningSolutions.Data.Models.TutorialContent;
     using DigitalLearningSolutions.Web.Helpers;
     using Microsoft.Extensions.Configuration;
 
     public class ContentViewerViewModel
     {
-        public string HtmlSource { get; }
-        public string ScormSource { get; }
+        public int CustomisationId { get; }
+        public int CentreId { get; }
+        public int SectionId { get; }
+        public int TutorialId { get; }
+        public int CandidateId { get; }
+        public int ProgressId { get; }
+        public string TutorialName { get; }
+        public string CourseTitle { get; }
+        public string ContentSource { get; }
 
-        public ContentViewerViewModel(IConfiguration config)
+        private static readonly Regex ScormRegex = new Regex(@".*imsmanifest\.xml$");
+
+        public ContentViewerViewModel(
+            IConfiguration config,
+            TutorialContent tutorialContent,
+            int customisationId,
+            int centreId,
+            int sectionId,
+            int tutorialId,
+            int candidateId,
+            int progressId
+        )
         {
-            HtmlSource = GetHtmlSource(config);
-            ScormSource = GetScormSource(config);
+            CustomisationId = customisationId;
+            CentreId = centreId;
+            SectionId = sectionId;
+            TutorialId = tutorialId;
+            CandidateId = candidateId;
+            ProgressId = progressId;
+
+            TutorialName = tutorialContent.TutorialName;
+            CourseTitle = tutorialContent.CourseTitle;
+
+            if (IsScormPath(tutorialContent.TutorialPath!))
+            {
+                ContentSource = GetScormSource(config, tutorialContent);

```

Comment:
Can be slightly simplified using ternary operator 🙂

Reply:
Yep, done in 79d4b90 🙂